bin/improver-ensemble-calibration

```diff
@@ -123,34 +143,94 @@ def main():
+        current_forecast = (
+            RebadgePercentilesAsRealizations().process(current_forecast))
+
+    # If percentiles, rebadge as realizations.
+    try:
+        find_percentile_coordinate(current_forecast)
+        input_forecast_type = "percentiles"
+    except CoordinateNotFoundError:
+        pass
+    if input_forecast_type == "percentiles":
+        if not args.num_realizations:
+            raise ValueError(msg.format(input_forecast_type))
+        # Resample percentiles, so that the percentiles are equally spaced.
+        current_forecast = ResamplePercentiles(
+            ecc_bounds_warning=args.ecc_bounds_warning).process(
+                current_forecast, no_of_percentiles=args.num_realizations)
+        current_forecast = (
+            RebadgePercentilesAsRealizations().process(current_forecast))
+
     # Default number of ensemble realizations is the number in
     # the raw forecast.
     if not args.num_realizations:
         args.num_realizations = len(
             current_forecast.coord('realization').points)
 
     # Ensemble-Calibration to calculate the mean and variance.
-    forecast_predictor_and_variance = EnsembleCalibration(
+    forecast_predictor, forecast_variance = EnsembleCalibration(
         args.calibration_method, args.distribution, args.units,
         predictor_of_mean_flag=args.predictor_of_mean).process(
             current_forecast, historic_forecast, truth)
+
     # If required, save the mean and variance.
-    if args.save_mean_variance:
-        mean_variance = [x for y in forecast_predictor_and_variance for x in y]
-        mean_variance = iris.cube.CubeList(mean_variance)
-        save_netcdf(mean_variance, args.save_mean_variance)
-
-    # Ensemble-Copula-Coupling to generate realizations from mean and variance.
-    percentiles = GeneratePercentilesFromMeanAndVariance().process(
-        forecast_predictor_and_variance, args.num_realizations)
-    result = EnsembleReordering().process(percentiles, current_forecast,
-                                          random_ordering=args.random_ordering,
-                                          random_seed=args.random_seed)
+    if args.save_mean:
+        save_netcdf(forecast_predictor.copy(), args.save_mean)
+    if args.save_variance:
+        save_netcdf(forecast_variance.copy(), args.save_variance)
+
+    # If input forecast is probabilities, convert output into probabilities.
+    # If input forecast is percentiles, convert output into percentiles.
+    # If input forecast is realizations, convert output into realizations.
+    if input_forecast_type == "probabilities":
+        result = GenerateProbabilitiesFromMeanAndVariance().process(
+            forecast_predictor, forecast_variance, original_current_forecast)
+    elif input_forecast_type == "percentiles":
+        result = GeneratePercentilesFromMeanAndVariance().process(
+            forecast_predictor, forecast_variance, args.num_realizations)
```

Comment:
Should this output at the same percentiles as the input cube rather than evenly spaced ones?

Reply:
As suggested, I've made the output percentiles from the CLI match the percentiles that are input, which I think is the desired behaviour.